### ISSUE TITLE
github.com Enterprise runners advice

### ISSUE THREAD

**Issue by dhrapson** [vbem/multi-runners#17]
Hi,
Can you advise whether its possible to deploy a GitHub.com Enterprise runners as described on https://docs.github.com/en/enterprise-cloud@latest/admin/managing-github-actions-for-your-enterprise/getting-started-with-github-actions-for-your-enterprise/getting-started-with-self-hosted-runners-for-your-enterprise? 
Or whether you plan to add support for that pattern?

By this I mean a single Runner in a Runner Group that exists at the GH Enterprise, and the group is shared across multiple Orgs.
I use [GitHub Enterprise Cloud](https://docs.github.com/en/enterprise-cloud@latest/admin/overview/about-github-enterprise-cloud) and deploying runner groups per org is considerable duplication, where we have a number of Orgs, so I was interested to know if the above is in your plans

**Comment by vbem** [comment]
Hi,

Thanks for your fantastic advice.

I checked the [relevant API documents](https://docs.github.com/en/enterprise-cloud@latest/rest/actions/self-hosted-runners?apiVersion=2022-11-28#create-a-registration-token-for-an-enterprise) of GHEC (GitHub Enterprise Cloud), although there are some API difference, it's possible for this wrapper script to support GHEC enterprise level runners registration, that is, to register runners for all orgs/repos within an enterprise.

However, currently I don't have an testing environment of GHEC. And adding enterprise level support may needs some refactoring efforts on current wrapper script. I'll try to add this feature according to GHEC API document later.

I'll also keep this issue open until this feature was added.
